packages/react-dom/src/client/ReactInputSelection.js

```diff
@@ -26,7 +26,12 @@ export function hasSelectionCapabilities(elem) {
   const nodeName = elem && elem.nodeName && elem.nodeName.toLowerCase();
   return (
     nodeName &&
-    ((nodeName === 'input' && elem.type === 'text') ||
+    ((nodeName === 'input' &&
+      (elem.type === 'text' ||
+        elem.type === 'search' ||
+        elem.type === 'tel' ||
+        elem.type === 'url' ||
+        elem.type === 'password')) ||
```

Comment:
Why is email not in this list?

Reply:
Sorry. That was very blunt! Should `email` be in here? I found a neat table from a spec:
https://html.spec.whatwg.org/multipage/input.html#concept-input-apply

So this matches up with the `setSelectionRange` column? Dang. I really want to know the reasoning behind not supporting this for all text based input. 

Would it be possible to add a comment at the top of this block that references this table for future contributors?

Reply:
I will add a comment containing the link as well. As far as I can see we are only using the [`selectionStart`](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactInputSelection.js#L91) and [`selectionEnd`](https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactInputSelection.js#L117) in the code. So I guess we match up with these two rows.